src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -134,7 +213,32 @@ internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
             out int newOffset,
             uint sequenceNumber)
         {
-            throw new PlatformNotSupportedException();
+            if (offset < 0 || offset > (buffer == null ? 0 : buffer.Length))
```

Comment:
I didn't realize this was copied from the Windows code.  Let's be careful here... does your "fixed" change it to something that could end up throwing an exception out of public API surface area in a place where it didn't previously?  If so, we need to revert that.